sdk/iot/Azure.Iot.Hub.Service/src/API Design.md

```diff
@@ -107,7 +107,65 @@ APIs for managing configurations for devices and modules.
+/// <param name="configuration">Twin configuration to update</param>
+/// <param name="precondition">The condition on which to perform this operation</param>
+/// <param name="cancellationToken">The cancellation token.</param>
+/// <returns>TwinConfiguration for a automatic device/module management</returns>
+public virtual async Task<Response<TwinConfiguration>> UpdateConfigurationAsync(string configurationId, TwinConfiguration configuration, IfMatchPrecondition precondition = IfMatchPrecondition.IfMatch, CancellationToken cancellationToken = default)
+
+/// <summary>
+/// Deletes a configuration on the IoT Hub for automatic device/module management
+/// </summary>
+/// <param name="configurationId">The unique identifier of the configuration.</param>
+/// <param name="precondition">The condition on which to perform this operation</param>
+/// <param name="cancellationToken">The cancellation token.</param>
+/// <returns>The Http Response</returns>
+public virtual async Task<Response> DeleteConfigurationAsync(string configurationId, IfMatchPrecondition precondition = IfMatchPrecondition.IfMatch, CancellationToken cancellationToken = default)
+
+/// <summary>
+/// Gets configurations on the IoT Hub for automatic device/module management. Pagination is not supported.
+/// </summary>
+/// <param name="count">The number of configurations to retrieve. TODO: Can value be overriden if too large?.</param>
+/// <param name="cancellationToken">The cancellation token.</param>
+/// <returns>A list of TwinConfiguration for a automatic device/module management</returns>
+public virtual async Task<Response<IReadOnlyList<TwinConfiguration>>> GetConfigurationsAsync(int? count = null, CancellationToken cancellationToken = default)
+
+/// <summary>
+/// Validates target condition and custom metric queries for a configuration on the IoT Hub.
+/// </summary>
+/// <param name="configuration">The configuration for target condition and custom metric queries.</param>
+/// <param name="cancellationToken">The cancellation token.</param>
+/// <returns>TwinConfiguration for a automatic device/module management</returns>
+public virtual async Task<Response<ConfigurationQueriesTestResponse>> TestQueriesAsync(ConfigurationQueriesTestInput configuration, CancellationToken cancellationToken = default)
+
+/// <summary>
+/// Applies the provided configuration content to the specified edge device.
+/// </summary>
+/// <param name="id">The unique identifier of the edge device.  TODO <service team>: Is it just device or edge device?".</param>
+/// <param name="content">Configuration Content. TODO <service team>: Get more context</param>
+/// <param name="cancellationToken">The cancellation token.</param>
+/// <returns>TODO <service team>: fix to return empty response</returns>
+public virtual async Task<Response<object>> ApplyOnEdgeDeviceAsync(string id, ConfigurationContent content, CancellationToken cancellationToken = default)
```

Comment:
We could make the parameter "deviceId" instead of "id" to make it clear what the id identifies

Reply:
This isn't resolved. I still see the parameter named "id" instead of "deviceId"

Reply:
my bad, somewhere got reverted when I had accidently checked in other changes. Let me fix it